server/src/lib/challenge.ts

```diff
@@ -0,0 +1,104 @@
+import { Request, Response } from 'express';
+import * as fs from 'fs';
+
+const PromiseRouter = require('express-promise-router');
+
+// [TODO](can) mock data for challenge board, remove it when implementing detailes of the challenge APIs.
+const getMockData = () => {
+  let data = null;
+  try {
+    data = JSON.parse(fs.readFileSync('./challenge-mock-data.json', 'utf-8'));
+  } catch (err) {
+    console.error(err, 'could not load config.json, using defaults');
+  }
+  return data;
+};
+
+export default class Challenge {
+  getRouter() {
+    const router = PromiseRouter({ mergeParams: true });
+
+    router.get('/:challenge/points', this.getPoints);
+    router.get('/:challenge/progress', this.getWeeklyProgress);
+    router.get('/:challenge/:locale/members/:type', this.getTopMembers);
+    router.get('/:challenge/:locale/teams', this.getTopTeams);
+    router.get(
+      '/:challenge/:locale/contributors/:type',
+      this.getTopContributors
+    );
+
+    return router;
+  }
+
+  getPoints = async (
+    { client_id, params: { challeng_token } }: Request,
```

Comment:
Can we update the spelling here `challeng_token` => `challenge_token`? Where is "challeng" coming from?

Reply:
A typo, my fault, should be challenge_token. Just not sure if
front end have that value.